### DISCUSSION TITLE
Conditional Validation

### DISCUSSION THREAD

**Discussion by krunaldodiya** [adonisjs/core#3243] [Help]
Hello,

How can I use conditional validation in Adonis 5 ?

like 

```
schema.create({
  avatar: schema.string({ trim: true }) || schema.file({ extnames: ['jpg', 'png'] })
})
```

**Comment by leognmotta** [comment]
I am not sure how you can do it, but you can always create custom validation and perform logic on it.

[https://docs.adonisjs.com/guides/validator/custom-rules](https://docs.adonisjs.com/guides/validator/custom-rules)
